internal/cluster/validations/validations.go

```diff
@@ -905,3 +912,27 @@ func ValidateHighAvailabilityModeWithPlatform(highAvailabilityMode *string, plat
 
 	return nil
 }
+
+func ValidateIgnitionImageSize(config string) error {
+	var err error
+	var data *bytes.Reader
+	var ignitionBytes []byte
+
+	// Ensure that the ignition archive isn't larger than 256KiB
+	configBytes := []byte(config)
+	content := isoeditor.IgnitionContent{Config: configBytes}
+	data, err = content.Archive()
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to create ignition archive (config: %s)", config))
+	}
+	ignitionBytes, err = io.ReadAll(data)
```

Comment:
Why re-reading all bytes as opposed as `data.Len()`?

Reply:
Good catch, should be enough to get the length of unread bytes. Changed.